src/site_pattern.hpp

```diff
@@ -15,8 +15,8 @@
 class SitePattern {
  public:
   SitePattern() = default;
-  explicit SitePattern(const Alignment& alignment, const TagStringMap& tag_taxon_map)
-      : alignment_(alignment), tag_taxon_map_(tag_taxon_map) {
+  SitePattern(const Alignment& alignment, TagStringMap tag_taxon_map)
```

Comment:
Better change tag_taxon_map to be rvalue (TagStringMap&& tag_taxon_map), since it will be moved from.

Reply:
OK, I'm glad to have you to talk this over. See 

https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference

Reply:
You're right - my comment is not valid. I assumed the constructor is meant to only accept rvalues, but it's used with lvalues in the rest of the code.